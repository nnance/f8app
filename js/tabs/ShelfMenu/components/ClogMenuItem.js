import React from 'react';
import {
  View,
  Image,
  Text,
  TouchableOpacity,
} from 'react-native';

import WhiteMenuItem from './WhiteMenuItem';
import styles from './styles';

// cant be stateless component
/* eslint react/no-multi-comp: warn */

export class ShowTagButton extends React.Component {
  render() {
    const { onPress } = this.props;
    return (<TouchableOpacity onPress={onPress}>
      <Image source={require('../../../assets/shelf-menu/show-tag.png')} style={{ width: 20, height: 20, resizeMode: 'contain' }} />
    </TouchableOpacity>);
  }
}

export class HideTagButton extends React.Component {
  render() {
    const { onPress } = this.props;
    return (<TouchableOpacity onPress={onPress}>
      <Image source={require('../../../assets/shelf-menu/hide-tag.png')} style={{ width: 20, height: 20, resizeMode: 'contain' }} />
    </TouchableOpacity>);
  }
}

class ClogMenuItem extends React.Component {
  constructor(...args) {
    super(...args);
    this.state = {
      showTag: false,
    };

    this.onShowTagPress = this.onShowTagPress.bind(this);
    this.onHideTagPress = this.onHideTagPress.bind(this);
  }

  onShowTagPress() {
    this.setState({
      showTag: true,
    });
  }

  onHideTagPress() {
    this.setState({
      showTag: false,
    });
  }

  onTagPress(id) {
    if (this.props.onTagPress) {
      this.props.onTagPress(id);
    }
  }

  renderTagItem() {
    const tags = this.props.tags;
    return (
      <View>
        {
          tags.map(tag => (
            <TouchableOpacity key={tag.id} onPress={() => this.onTagPress(tag.id)}>
              <Text style={[styles.fadedWhiteText, { paddingVertical: 5 }]}>{tag.title}</Text>
            </TouchableOpacity>
          ))
        }
      </View>
    );
  }

  render() {
    return (
      <View>
        <WhiteMenuItem
          style={{ paddingTop: 10, paddingBottom: this.state.showTag ? 5 : 10 }}
          title={this.props.title}
          onPress={this.props.onClogPress}
          icon={<Image source={this.props.source} style={{ width: 25, height: 25, resizeMode: 'contain' }} />}
          renderedButton={
            !this.state.showTag ?
              <ShowTagButton onPress={this.onShowTagPress} />
            : <HideTagButton onPress={this.onHideTagPress} />
          }
        />
        {
          this.state.showTag ?
            <View style={{ paddingLeft: 40, marginBottom: 10 }}>{this.renderTagItem()}</View>
            : null
        }
        {
          !this.props.notShowBottomLine ?
            <View style={{ flex: 1, marginVertical: 0, marginLeft: 40, height: 1, backgroundColor: 'rgba(0, 0, 0, 0.2)' }} />
            : null
        }
      </View>
    );
  }
}

export default ClogMenuItem;
