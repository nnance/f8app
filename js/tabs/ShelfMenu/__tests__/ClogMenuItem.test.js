import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { TouchableOpacity } from 'react-native';

import ClogMenuItem, { HideTagButton, ShowTagButton } from '../components/ClogMenuItem';

const user = {
  name: 'ima',
};

const tags = [{ id: '1', title: 'Chat Clog' }, { id: '2', title: 'Video Clog' }, { id: '3', title: 'Video Clog' }];

describe('ClogMenuItem', () => {
  it('render', () => {
    const tree = renderer.create(<ClogMenuItem tags={tags} user={user} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('render with show/hide sub item', () => {
    const wrapper = shallow(<ClogMenuItem tags={tags} user={user} />);
    wrapper.instance().onShowTagPress();
    wrapper.update();
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.instance().onHideTagPress();
    wrapper.update();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('call onTagPress on press sub item', () => {
    const spy = jest.fn();
    const dump = shallow(<ClogMenuItem tags={tags} user={user} onTagPress={spy} />);
    const wrapper = shallow(dump.instance().renderTagItem());
    wrapper.find(TouchableOpacity).at(1).simulate('press');
    expect(spy).toBeCalledWith('2');
  });

  it('HideTagButton render', () => {
    const tree = renderer.create(<HideTagButton />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('ShowTagButton render', () => {
    const tree = renderer.create(<ShowTagButton />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
