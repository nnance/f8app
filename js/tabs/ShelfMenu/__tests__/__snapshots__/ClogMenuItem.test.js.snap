exports[`ClogMenuItem HideTagButton render 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}>
  <Image
    source={1}
    style={
      Object {
        "height": 20,
        "resizeMode": "contain",
        "width": 20,
      }
    } />
</View>
`;

exports[`ClogMenuItem ShowTagButton render 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}>
  <Image
    source={1}
    style={
      Object {
        "height": 20,
        "resizeMode": "contain",
        "width": 20,
      }
    } />
</View>
`;

exports[`ClogMenuItem render 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingLeft": 5,
          "paddingVertical": 10,
        },
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        },
      ]
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "opacity": 1,
        }
      }
      testID={undefined}>
      <View
        style={
          Object {
            "height": 25,
            "width": 25,
          }
        }>
        <Image
          source={undefined}
          style={
            Object {
              "height": 25,
              "resizeMode": "contain",
              "width": 25,
            }
          } />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 10,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          } />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "justifyContent": "center",
          "marginRight": 10,
          "width": 50,
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}>
        <Image
          source={1}
          style={
            Object {
              "height": 20,
              "resizeMode": "contain",
              "width": 20,
            }
          } />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "flex": 1,
        "height": 1,
        "marginLeft": 40,
        "marginVertical": 0,
      }
    } />
</View>
`;

exports[`ClogMenuItem render with show/hide sub item 1`] = `
<View>
  <WhiteMenuItem
    icon={
      <Image
        style={
          Object {
            "height": 25,
            "resizeMode": "contain",
            "width": 25,
          }
        } />
    }
    renderedButton={
      <HideTagButton
        onPress={[Function]} />
    }
    style={
      Object {
        "paddingBottom": 5,
        "paddingTop": 10,
      }
    } />
  <View
    style={
      Object {
        "marginBottom": 10,
        "paddingLeft": 40,
      }
    }>
    <View>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgba(255, 255, 255, 0.2)",
              },
              Object {
                "paddingVertical": 5,
              },
            ]
          }>
          Chat Clog
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgba(255, 255, 255, 0.2)",
              },
              Object {
                "paddingVertical": 5,
              },
            ]
          }>
          Video Clog
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgba(255, 255, 255, 0.2)",
              },
              Object {
                "paddingVertical": 5,
              },
            ]
          }>
          Video Clog
        </Text>
      </TouchableOpacity>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "flex": 1,
        "height": 1,
        "marginLeft": 40,
        "marginVertical": 0,
      }
    } />
</View>
`;

exports[`ClogMenuItem render with show/hide sub item 2`] = `
<View>
  <WhiteMenuItem
    icon={
      <Image
        style={
          Object {
            "height": 25,
            "resizeMode": "contain",
            "width": 25,
          }
        } />
    }
    renderedButton={
      <ShowTagButton
        onPress={[Function]} />
    }
    style={
      Object {
        "paddingBottom": 10,
        "paddingTop": 10,
      }
    } />
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "flex": 1,
        "height": 1,
        "marginLeft": 40,
        "marginVertical": 0,
      }
    } />
</View>
`;
